controllers/hostedcontrolplane/hostedcontrolplane.go

```diff
@@ -96,7 +96,7 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 
 	// Fetch the HostedControlPlane instance
 	hostedcontrolplane := &hypershiftv1beta1.HostedControlPlane{}
-	err := r.Client.Get(ctx, req.NamespacedName, hostedcontrolplane)
+	err := r.Get(ctx, req.NamespacedName, hostedcontrolplane)
```

Comment:
I don't see any changes to hostedcontrolplane struct. https://github.com/openshift/route-monitor-operator/blob/master/controllers/hostedcontrolplane/hostedcontrolplane.go#L74
Could you please explain this change, why we don't need the Client here?

Reply:
Yeah!  That's something new I learned about from the linter too!  Since the Client is an embedded struct, the *HostedControlPlaneReconciler automatically gets access to the Client's fields, acting like they are directly owned.  

eg:

```
type Job struct {
    title     string
    ID        int
 }

type Person struct {
   name string
   job      Job
}


// Allows for:

chris = Person{
   name: chris
   job: Job{
      title: SRE,
      ID: 1
    }
}

fmt.Println(chris.title)

// Prints: "SRE"
```